contrib/scalar/scalar.c

```diff
@@ -244,6 +277,9 @@ static int register_dir(void)
 	if (!res)
 		res = toggle_maintenance(1);
 
+	if (!res)
+		res = start_fsmonitor_daemon();
```

Comment:
Hmm. The longer I think about this, the more I believe that we should _not_ start the FSMonitor automatically.

Take the upgrade story, for example. Imagine a user upgrades Git for Windows to a version that includes (native) Scalar, and re-registers all Scalar enlistments with that native Scalar. Essentially, what it will run is:

```sh
git for-each-repo --config=scalar.repos -c alias.scalar='!scalar' scalar register
```

(I hope that we can find consensus that this is a horrible interface, and that we should implement something like `scalar upgrade-config --all` that does this instead. But that's beside the point.)

Now _all_ the enlistments will have running FSMonitor daemons, even if the user might not even need to work in most of those enlistments for the day. Even worse, those daemons don't do any good anyway because upon the first `git status`, they will still say "I have no information, you will have to fall back to a full scan.".

So maybe we can agree that we should _not_ start the built-in FSMonitor upon `scalar register`?

Reply:
Or maybe the better path forward is to add the `upgrade-config` verb that specifically does _not_ start the built-in FSMonitor, but does all the rest of what `register` would do.

Yeah, I think I like that one better. Thoughts?

Reply:
The `upgrade-config` step would be a post-install step, so it would be appropriate to launch a new FS Monitor daemon and even update the index with a new token to ensure the user has a fast first interaction after upgrade.

But this can be handled in a future release. For now, perhaps focus on setting the config and ensuring that the daemon is started when the user starts interacting with the repo.